dali/util/cucontext.h

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) 2017-2018, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_UTIL_CUCONTEXT_H_
+#define DALI_UTIL_CUCONTEXT_H_
+
+#include <cuda.h>
+
+namespace dali {
+
+class CUContext {
+  public:
+    CUContext();
+    CUContext(CUdevice device, unsigned int flags = 0);
+    CUContext(CUcontext);
```

Comment:
If this contructor takes ownership of the CUcontext, it should be clearly indicated in a comment. Also, it should be explicit, as are constructors of smart pointers.

Reply:
Ok we can add a comment here, but how to make it more explicit than with a comment?